### PR TITLE
Correct clippy complaints

### PR DESCRIPTION
**Pull request by budziq**
I've run clippy over the code and corrected all of its warnings.

Transition from [`ok_or`](https://doc.rust-lang.org/std/option/enum.Option.html#method.ok_or) to [`ok_or_else`](https://doc.rust-lang.org/std/option/enum.Option.html#method.ok_or_else) might be non obvious but the first one calls the "or" branch unconditionally (suggested only if Err value is already available) while the second one evaluates it lazily.

### CODE REVIEW

File: src/renderer/html_handlebars/hbs_renderer.rs [hunk 1]
```diff
@@ -121,12 +121,10 @@ impl HtmlHandlebars {
...
-        let rendered = build_header_links(rendered, "print.html");
-        let rendered = fix_anchor_links(rendered, "print.html");
-        let rendered = fix_code_blocks(rendered);
-        let rendered = add_playpen_pre(rendered);
-
-        rendered
+        let rendered = build_header_links(&rendered, "print.html");
+        let rendered = fix_anchor_links(&rendered, "print.html");
+        let rendered = fix_code_blocks(&rendered);
+        add_playpen_pre(&rendered)
```
Comment (azerupi) [inline src/renderer/html_handlebars/hbs_renderer.rs]: I understand why Clippy suggests this, but I am of the opinion that the previous code is actually somewhat clearer and more coherent in this case. No?

Comment (budziq) [inline src/renderer/html_handlebars/hbs_renderer.rs]: Well some of the changes might be arguable but these things tend to be subjective. I'm not really attached to this work As I mostly wanted to take clippy and rustfix for a spin. Just tell me which changes you don't like and I'll drop these :)

**Comment by budziq** [comment]
> could you run rustfmt on the changes perhaps?

Gladly but I would rather wait for some of the PR's to be merged first I would not like to make spam all other contributors with merge conflicts ;)

File: src/renderer/html_handlebars/hbs_renderer.rs (hunk 1 above)
Comment (azerupi) [inline src/renderer/html_handlebars/hbs_renderer.rs]: Just this one, the rest looks good 👍

**Comment by azerupi** [comment]
Yes, indeed. In that case let's merge this and make a new rustfmt PR that we can merge when the PR count is a little lower :)

**Comment by budziq** [comment]
updated :)

**Comment by azerupi** [comment]
Thanks!

**Pull request merged**